--- Pyrex-0.9.4/Pyrex/Distutils/build_ext.py.orig	2006-04-17 16:33:06.000000000 +0200
+++ Pyrex-0.9.4/Pyrex/Distutils/build_ext.py	2006-04-17 16:33:39.000000000 +0200
@@ -65,8 +65,7 @@
     return [replace_suffix(src, suffix) for src in pyx_sources] + other_sources
 
   def pyrex_compile(self, source):
-  	options = CompilationOptions(default_options,
-  		include_path = self.include_dirs)
+    options = CompilationOptions(default_options,include_path = self.include_dirs)
     result = compile(source, options)
     if result.num_errors <> 0:
       sys.exit(1)
